cmd/main.go

```diff
@@ -15,8 +15,16 @@ import (
 	"github.com/buildwithgrove/path/router"
 )
 
-// TODO_UPNEXT(@adshmh): add CLI flags for specifying the config file's path.
-var configPath = filepath.Join("config", ".config.yaml")
+// Define a variable to hold the config path
+var configPath string
+
+const defaultConfigPath = "/app/.config.yaml"
```

Comment:
> @commoddity It's still unclear to me if we are sticking with `/app/config/.config.yaml` or `/app/.config.yaml`.
> 
> This will require changes in poktroll that we merged in recently.
> 
> Why the change back?

The only reason is that the mount path for the secrets file in `./local/kubernetes/path-values.yaml` was using `/app/.config.yaml`.

It's fine to update both the hardcoded default in main.go and the mountPath in the kubernetes yaml file to `/app/config/.config.yaml`. I'll do that.

Also want to point out that this implements Arash's TODO to allow configuration of the config YAML location with a command line flag.

For example, the make path_run target now runs the binary like: `./path -config ./config/.config.yaml` as the `.config.yaml` file is copied to `./bin/config/.config.yaml` by the `make copy_*_config` targets.

Reply:
UPDATE: Actually after changing the kubernetes `mountPath` to `/app/config/.config.yaml` the PATH container breaks. I've spent about an hour trying to get it to work without success; don't want to burn too much time on it so tl;dr: 
- `mountPath: /app/config/.config.yaml` - DOES NOT WORK
```bash
Error: failed to create containerd task: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error mounting "/var/lib/kubelet/pods/d49efd48-4ecb-4191-9068-b4363eb6f5ff/volume-subpaths/path-config-local/path/1" to rootfs at "/app/config/.config.yaml": mount /var/lib/kubelet/pods/d49efd48-4ecb-4191-9068-b4363eb6f5ff/volume-subpaths/path-config-local/path/1:/app/config/.config.yaml (via /proc/self/fd/6), flags: 0x5001: not a directory: unknown
```
- `mountPath: /app/.config.yaml` - WORKS

We can either recruit Hebert to understand why or just use `/app/.config.yaml` as the mount path.

The line in question is: https://github.com/buildwithgrove/path/blob/envoy-auth-tiltfile/local/kubernetes/path-values.yaml#L10

Reply:
The error you're seeing is related to how the deployment is trying locate and mount the configuration. This means that `/app/config/.config.yaml` is not defined in the [mount configuration](https://github.com/buildwithgrove/path/blob/envoy-auth-tiltfile/local/kubernetes/path-values.yaml#L7-L15).

In order to fix this, we just need to update the configuration to look like this:

```yaml
path:
  mountSecrets:
    - name: path-config-local
      mountPath: /app/config/.config.yaml
      subPath: .config.yaml
      items:
        - key: .config.yaml
          path: .config.yaml
```

Give it a try and let me know if it works.

Reply:
> The error you're seeing is related to how the deployment is trying locate and mount the configuration. This means that `/app/config/.config.yaml` is not defined in the [mount configuration](https://github.com/buildwithgrove/path/blob/envoy-auth-tiltfile/local/kubernetes/path-values.yaml#L7-L15).
> 
> In order to fix this, we just need to update the configuration to look like this:
> 
> ```yaml
> path:
>   mountSecrets:
>     - name: path-config-local
>       mountPath: /app/config/.config.yaml
>       subPath: .config.yaml
>       items:
>         - key: .config.yaml
>           path: .config.yaml
> ```
> 
> See my comment in the [channel](https://discord.com/channels/824324475256438814/1273320783547990160/1316144188718583919) for further explanation.
> 
> Give it a try and let me know if it works.

That's what I did to cause the error.